### PR TITLE
Fix restoring symlink timestamps for linux

### PR DESCRIPTION
**Pull request by fd0**
This adds a test and fixes restoring symlink timestamps for Linux. We need to figure out how to do this on darwin and *bsd.

### CODE REVIEW

File: node_linux.go [hunk 1]
```diff
@@ -22,3 +24,36 @@ func (node *Node) fillTimes(stat *syscall.Stat_t) {
...
+
+const AT_SYMLINK_NOFOLLOW = 0x100
+
+func utimensat(dirfd int, path string, times *[2]syscall.Timespec, flags int) (err error) {
+	var _p0 *byte
+	_p0, err = syscall.BytePtrFromString(path)
+	if err != nil {
+		return
+	}
+	_, _, e1 := syscall.Syscall6(syscall.SYS_UTIMENSAT, uintptr(dirfd), uintptr(unsafe.Pointer(_p0)), uintptr(unsafe.Pointer(times)), uintptr(flags), 0, 0)
```
Comment (fw42) [inline node_linux.go]: this looks scary. do we really want to do this?

Comment (fd0) [inline node_linux.go]: I'm not sure yet. It's the same they do in the stdlib for Go 1.5...

File: node.go [hunk 2]
```diff
@@ -166,12 +164,20 @@ func (node Node) restoreMetadata(path string) error {
...
-	var utimes = []syscall.Timespec{
+	var utimes = [...]syscall.Timespec{
 		syscall.NsecToTimespec(node.AccessTime.UnixNano()),
 		syscall.NsecToTimespec(node.ModTime.UnixNano()),
 	}
 
-	if err := syscall.UtimesNano(path, utimes); err != nil {
+	if node.Type == "symlink" {
+		if err := node.restoreSymlinkTimestamps(path, utimes); err != nil {
+			return errors.Annotate(err, "UtimesNano")
```
Comment (fw42) [inline node.go]: I feel like this annotation should happen in `restoreSymlinkTimestamps`, not here

File: node_linux.go (hunk 1 above)
Comment (fw42) [inline node_linux.go]: why do you need this `e1`? why not overwrite the other `err`?

File: node_linux.go [hunk 3]
```diff
@@ -22,3 +24,36 @@ func (node *Node) fillTimes(stat *syscall.Stat_t) {
...
+	if e1 != 0 {
+		err = e1
+	}
+	return
+}
+
+//sys	utimensat(dirfd int, path string, times *[2]Timespec, flags int) (err error)
```
Comment (fw42) [inline node_linux.go]: I don't think we need this comment

**Comment by fw42** [comment]
I guess this works, but I'm not a big fan. I hope we can get rid of this soon and use something from the standard library.

File: node_linux.go (hunk 3 above)
Comment (fd0) [inline node_linux.go]: I've copied that from the generated documentation, but yeah, isn't necessary.

File: node.go (hunk 2 above)
Comment (fd0) [inline node.go]: I agree.

**Comment by fd0** [comment]
Yeah, me neither. I've fixed the error annotation and will merge this once travis agrees.

**Pull request merged**